hazelcast/src/main/java/com/hazelcast/query/impl/AttributeIndexRegistry.java

```diff
@@ -186,7 +190,12 @@ public String getName() {
 
         @Override
         public String[] getComponents() {
-            return null;
+            return components;
+        }
+
+        @Override
+        public IndexConfig getConfig() {
+            return delegate.getConfig();
```

Comment:
Probably it's better to thrown an exception here, like it's done in `getName`. The index config is not really defining the decorated index.

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/15537/commits/2163cbdd5ac5290a80317aa92679e8ded7520b42